plugins/vpp/abfplugin/vppcalls/vpp1908/abf_vppcalls.go

```diff
@@ -126,26 +136,46 @@ func (h *ABFVppHandler) toFibPaths(abfPaths []*vpp_abf.ABF_ForwardingPath) (fibP
 			SwIfIndex:  ifData.SwIfIndex,
 			Weight:     uint8(abfPath.Weight),
 			Preference: uint8(abfPath.Preference),
-			IsDvr:      boolToUint(abfPath.Dvr),
-		}
-
-		// next hop IP
-		nextHop := net.ParseIP(abfPath.NextHopIp)
-		if nextHop.To4() == nil {
-			nextHop = nextHop.To16()
-			fibPath.Afi = 1
-		} else {
-			nextHop = nextHop.To4()
-			fibPath.Afi = 0
+			Type:       setFibPathType(abfPath.Dvr),
 		}
-		fibPath.NextHop = nextHop
-
+		fibPath.Nh, fibPath.Proto = setFibPathNhAndProto(abfPath.NextHopIp)
 		fibPaths = append(fibPaths, fibPath)
 	}
 
 	return fibPaths
 }
 
+// supported cases are DVR and normal
+func setFibPathType(isDvr bool) abf.FibPathType {
+	if isDvr {
+		return abf.FIB_API_PATH_TYPE_DVR
+	}
+	return abf.FIB_API_PATH_TYPE_NORMAL
+}
+
+// resolve IP address and return FIB path next hop (IP address) and IPv4/IPv6 version
+func setFibPathNhAndProto(ipStr string) (nh abf.FibPathNh, proto abf.FibPathNhProto) {
+	netIP := net.ParseIP(ipStr)
+	if netIP == nil {
+		return
```

Comment:
This does not handle invalid IP address. Error should be returned if parse IP is nil.